web-common/src/runtime-client/query-matcher.ts

```diff
@@ -0,0 +1,54 @@
+import type { Query } from "@tanstack/query-core";
+
+export enum QueryRequestType {
+  MetricsViewTopList = "toplist",
+  MetricsViewCompareTopList = "compare-toplist",
+  MetricsViewTimeSeries = "timeseries",
+  MetricsViewTotals = "totals",
+  MetricsViewRows = "rows",
+  ColumnRollupInterval = "rollup-interval",
+  ColumnTopK = "topk",
+  ColumnNullCount = "null-count",
+  ColumnDescriptiveStatistics = "descriptive-statistics",
+  ColumnTimeGrain = "smallest-time-grain",
+  ColumnNumericHistogram = "numeric-histogram",
+  ColumnRugHistogram = "rug-histogram",
+  ColumnTimeRange = "time-range-summary",
+  ColumnCardinality = "column-cardinality",
+  ColumnTimeSeries = "timeseries",
+  TableCardinality = "table-cardinality",
+  TableColumns = "columns-profile",
+  TableRows = "rows",
+}
+
+const TopLevelProfilingQuery: Partial<Record<QueryRequestType, boolean>> = {
+  [QueryRequestType.TableColumns]: true,
+  [QueryRequestType.TableRows]: true,
+};
```

Comment:
Does it make sense to include `TableCardinality` in this list? Then the variable could be `TableProfilingQuery` (not `TopLevelProfilingQuery`). It's not obvious to me why `TableColumns` and `TableRows` are in this unique group together.

Reply:
Yes updating